internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestLeaderReplicaFailover.java

```diff
@@ -0,0 +1,228 @@
+import com.linkedin.venice.utils.VeniceProperties;
+import com.linkedin.venice.writer.VeniceWriter;
+import com.linkedin.venice.writer.VeniceWriterFactory;
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.manager.zk.ZKHelixAdmin;
+import org.apache.helix.model.InstanceConfig;
+import org.rocksdb.ComparatorOptions;
+import org.rocksdb.util.BytewiseComparator;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+@Test(singleThreaded = true)
+public class TestLeaderReplicaFailover {
+  String stringSchemaStr = "\"string\"";
+  String valueSchemaStr =
+      "{\"type\":\"record\",\"name\":\"KeyRecord\",\"fields\":[{\"name\":\"name\",\"type\":\"string\"}]}";
+  AvroSerializer serializer = new AvroSerializer(AvroCompatibilityHelper.parse(stringSchemaStr));
+  AvroSerializer valueSerializer = new AvroSerializer(AvroCompatibilityHelper.parse(valueSchemaStr));
+
+  private static final int TEST_TIMEOUT = 360 * Time.MS_PER_SECOND;
+  private static final String[] CLUSTER_NAMES =
+      IntStream.range(0, 1).mapToObj(i -> "venice-cluster" + i).toArray(String[]::new);
+
+  private List<VeniceMultiClusterWrapper> childDatacenters;
+  private List<VeniceControllerWrapper> parentControllers;
+  private VeniceTwoLayerMultiColoMultiClusterWrapper multiColoMultiClusterWrapper;
+  private String clusterName;
+  private VeniceClusterWrapper clusterWrapper;
+
+  @BeforeClass
+  public void setUp() {
+    String stringSchemaStr = "\"string\"";
+    serializer = new AvroSerializer(AvroCompatibilityHelper.parse(stringSchemaStr));
+    Properties serverProperties = new Properties();
+    serverProperties.setProperty(ConfigKeys.SERVER_PROMOTION_TO_LEADER_REPLICA_DELAY_SECONDS, Long.toString(1));
+    serverProperties.put(LF_MODEL_DEPENDENCY_CHECK_DISABLED, "true");
+    serverProperties.put(ROCKSDB_PLAIN_TABLE_FORMAT_ENABLED, false);
+    serverProperties.put(LEADER_DOOM_NOTIFIER_TEST_ONLY_ENABLED, true);
+    serverProperties.put(
+        CHILD_DATA_CENTER_KAFKA_URL_PREFIX + "." + DEFAULT_PARENT_DATA_CENTER_REGION_NAME,
+        "localhost:" + Utils.getFreePort());
+    Properties controllerPropery = new Properties();
+    Properties parent = new Properties();
+    controllerPropery.put(DEFAULT_OFFLINE_PUSH_STRATEGY, OfflinePushStrategy.WAIT_N_MINUS_ONE_REPLCIA_PER_PARTITION);
+    parent.put(DEFAULT_PARTITION_SIZE, 10000);
+    multiColoMultiClusterWrapper = ServiceFactory.getVeniceTwoLayerMultiColoMultiClusterWrapper(
```

Comment:
Do we need multiple colo and multiple cluster setup here? I assume single cluster in single colo is good enough.

Reply:
Right no need to have multi-colo setup. I will make it a single colo/cluster.